src/card/styles.ts

```diff
@@ -2,6 +2,50 @@ import { css } from '@emotion/react';
 
 import { tokens } from '../tokens';
 
+import { HighlightCardProps } from './card-internal';
+
+const variantMap = {
+  green: {
+    backgroundColor: tokens.colors.green,
+  },
+  navy: {
+    backgroundColor: tokens.colors.navy,
+  },
+  red: {
+    backgroundColor: tokens.colors.red,
+  },
+  orange: {
+    backgroundColor: tokens.colors.orange,
+  },
+  orangeLight: {
+    backgroundColor: tokens.colors.orangeLight,
+  },
+  yellow: {
+    backgroundColor: tokens.colors.yellow,
+  },
+  blue: {
+    backgroundColor: tokens.colors.blue,
+  },
+  purple: {
+    backgroundColor: tokens.colors.purple,
+  },
+  purpleLight: {
+    backgroundColor: tokens.colors.purpleLight,
+  },
+  pink: {
+    backgroundColor: tokens.colors.pink,
+  },
+  greyLight: {
+    backgroundColor: tokens.colors.greyLight,
+  },
+  greyMedium: {
+    backgroundColor: tokens.colors.greyMedium,
+  },
+  white: {
+    backgroundColor: tokens.colors.white,
+  },
+};
```

Comment:
```suggestion
const backgroundColorMap = {
  green: tokens.colors.green,
  navy: tokens.colors.navy,
  red: tokens.colors.red,
  orange: tokens.colors.orange,
  orangeLight: tokens.colors.orangeLight,
  yellow: tokens.colors.yellow,
  blue: tokens.colors.blue,
  purple: tokens.colors.purple,
  purpleLight: tokens.colors.purpleLight,
  pink: tokens.colors.pink,
  greyLight: tokens.colors.greyLight,
  greyMedium: tokens.colors.greyMedium,
  white: tokens.colors.white,
};
```

and then below: `background-color: ${backgroundColorMap[highlightVariant]}`

Unless you expect to soon also make other properties depend on `highlightVariant`

Reply:
You could take it one step further, see below